File: frontend/types/event.ts
export type EventStatus = 'pending' | 'processing' | 'processed' | 'failed' | 'dead_letter';

export interface PlatformEvent {
  id: string;
  event_type: string;
  source: string;
  actor_id: string | null;
  entity_type: string | null;
  entity_id: string | null;
  payload: Record<string, unknown>;
  correlation_id: string;
  status: EventStatus;
  error_message: string | null;
  created_at: string;
  processed_at: string | null;
}

export interface EventListResponse {
  items: PlatformEvent[];
  total: number;
  page: number;
  page_size: number;
  total_pages: number;
}

export interface OutboxStats {
  pending: number;
  processing: number;
  processed: number;
  failed: number;
  dead_letter: number;
  total: number;
}

File: backend/app/schemas/event.py
"""
Event and Outbox Pydantic schemas.
"""

from __future__ import annotations

import uuid
from datetime import datetime
from typing import Any

from pydantic import BaseModel, ConfigDict, Field

from app.models.event import EventStatus


class EventResponse(BaseModel):
    """Normalized platform event item."""

    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    event_type: str
    source: str
    actor_id: uuid.UUID | None = None
    entity_type: str | None = None
    entity_id: str | None = None
    payload: dict[str, Any] = Field(default_factory=dict)
    correlation_id: str
    status: EventStatus
    error_message: str | None = None
    created_at: datetime
    processed_at: datetime | None = None


class EventListResponse(BaseModel):
    """Paginated list of platform events."""

    items: list[EventResponse]
    total: int
    page: int
    page_size: int
    total_pages: int


class OutboxStatsResponse(BaseModel):
    """Counts of outbox events by status."""

    pending: int = 0
    processing: int = 0
    processed: int = 0
    failed: int = 0
    dead_letter: int = 0
    total: int = 0


class RetryDeadLetterRequest(BaseModel):
    """Request to retry a dead-lettered outbox event."""

    outbox_id: uuid.UUID
